src/gg.regression.unity.bots/Runtime/Scripts/RGOverlay/RGSequenceEditor.cs

```diff
@@ -0,0 +1,314 @@
+            {
+                Destroy(AvailableSegmentsList.transform.GetChild(i).gameObject);
+            }
+        }
+
+        /**
+         * <summary>
+         * Save the current Sequence to disk
+         * </summary>
+         */
+        public void SaveSequence()
+        {
+            var addedSegments = _dropZone.GetChildren();
+            
+            foreach (var segment in addedSegments)
+            {
+                var path = segment.payload["path"];
+                var hasValue = Enum.TryParse(segment.payload["type"], out BotSequenceEntryType type);
+                if (!hasValue)
+                {
+                    continue;
+                }
+                
+                if (type == BotSequenceEntryType.Segment)
+                {
+                    CurrentSequence.AddBotSegmentPath(path);
+                }
+                else if (type == BotSequenceEntryType.SegmentList)
+                {
+                    CurrentSequence.AddBotSegmentListPath(path);
+                }
+                else
+                {
+                    Debug.LogError($"RGSequenceEditor could not add Bot Segment of type {type}");
+                }
+            }
+
+            CurrentSequence.name = NameInput.text;
+            CurrentSequence.description = DescriptionInput.text;
+            CurrentSequence.SaveSequenceAsJson();
+        }
+
+        /**
+         * <summary>
+         * Reset the `name`, `description`, and Available Segments to their defaults
+         * </summary>
+         */
+        public void ResetEditor()
+        {
+            if (NameInput != null)
+            {
+                NameInput.text = string.Empty;
+            }
+
+            if (DescriptionInput != null)
+            {
+                DescriptionInput.text = string.Empty;
+            }
+            
+            ClearAvailableSegments();
+            
+            _dropZone.ClearChildren();
+
+            SetCreateSequenceButtonEnabled(false);
+        }
+
+        /**
+         * <summary>
+         * Load all Segments from disk
+         * </summary>
+         */
+        public void ReloadAvailableSegments()
+        {
+            SearchInput.text = string.Empty;
+            _segmentEntries = BotSegment.LoadAllSegments();
+            CreateAvailableSegments(_segmentEntries);
+        }
+
+        /**
+         * <summary>
+         * When the search input's value changes, filter the Segments. If the search input is empty, show all Segments
+         * </summary>
+         * <param name="text">Search input text value</param>
+         */
+        public void OnSearchInputChange(string text)
+        {
+            if (string.IsNullOrEmpty(text))
+            {
+                CreateAvailableSegments(_segmentEntries);
+            }
+            else
+            {
+                _filteredSegmentEntries = _segmentEntries.Where(s => s.entryName.Contains(text.Trim())).ToList();
```

Comment:
From your loom video, it might be good to .lowercase() these strings so you can search without remembering capitalization

Reply:
good catch, added 👍🏻